packages/user-interface/src/hooks/useOgonePayment.tsx

```diff
@@ -0,0 +1,33 @@
+import {
+  GenerateOgonePaymentMutationVariables,
+  OgonePayment,
+  useGenerateOgonePaymentMutation,
+} from "@nl-portal/nl-portal-api";
+import { useState } from "react";
+import OgonePaymentForm from "../components/OgonePaymentForm";
+
+const useOgonePayment = () => {
+  const [showPaymentRedirectForm, setShowPaymentRedirectForm] = useState(false);
+  const [paymentData, setPaymentData] = useState<OgonePayment>();
+  const [mutateFunction] = useGenerateOgonePaymentMutation();
+
+  const startPayment = (
+    paymentRequestPayload: GenerateOgonePaymentMutationVariables,
+  ) => {
+    mutateFunction({
+      variables: { ...paymentRequestPayload },
+      onCompleted: (data) => {
+        setPaymentData(data.generateOgonePayment);
+        setShowPaymentRedirectForm(true);
+      },
+    });
+  };
+
+  const renderPaymentRedirectForm = () => {
+    return paymentData && <OgonePaymentForm {...paymentData} />;
+  };
+
+  return { startPayment, renderPaymentRedirectForm, showPaymentRedirectForm };
```

Comment:
Heb je niet genoeg aan "renderPaymentRedirectForm" deze geeft undefined terug als mutation nog niet voltooid is?

Reply:
Klopt inderdaad, zal ik ook aanpassen.